walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,158 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Set;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> ret = new ArrayList<TimeRange>();
+    if(request.getDuration() > 1440){
+        return ret;
+    }
+    if(events.size() == 0) {
+        ret.add(TimeRange.WHOLE_DAY);
+        return ret;
+    }
+    ArrayList<Event> otherMeetings;
+    if(request.getOptionalAttendees().size() >= 0 && request.getAttendees().size() == 0)
+        otherMeetings = doAttendeesHaveAnyOtherMeetings(events, request, true);
+    else{
+        otherMeetings = doAttendeesHaveAnyOtherMeetings(events, request, false);
+    }
+    if(otherMeetings.size() == 0) {
+        ret.add(TimeRange.WHOLE_DAY);
+        return ret;
+    }
+        ArrayList<TimeRange> timeranges =  new ArrayList<TimeRange>();
```

Comment:
timeRanges

Also ArrayList<TimeRange> timeRanges = otherMeetings.stream().map(Event::getWhen).collect(Collectors.asList());

I think there is a sort function as well for streams?